ArduPlane/tiltrotor.cpp

```diff
@@ -90,29 +90,48 @@ void QuadPlane::tiltrotor_continuous_update(void)
     tilt.current_throttle = constrain_float(motors_throttle,
                                             tilt.current_throttle-max_change,
                                             tilt.current_throttle+max_change);
-    
+
     /*
       we are in a VTOL mode. We need to work out how much tilt is
-      needed. There are 3 strategies we will use:
+      needed. There are 4 strategies we will use:
+
+      1a) In QAUTOTUNE mode the angle will be set to zero.
+      1b) In QACRO, QSTABILIZE and QHOVER modes with rc_fwd_thr_ch null, the angle will be set to zero.
 
-      1) in QSTABILIZE or QHOVER the angle will be set to zero. This
-         enables these modes to be used as a safe recovery mode.
+      2) In QACRO, QSTABILIZE and QHOVER modes with rc_fwd_thr_ch assigned to a valid RC channel:
+           The forward throttle is determined by RC input.
 
-      2) in fixed wing assisted flight or velocity controlled modes we
+      3) In fixed wing assisted flight or velocity controlled modes we
          will set the angle based on the demanded forward throttle,
          with a maximum tilt given by Q_TILT_MAX. This relies on
-         Q_VFWD_GAIN being set
+         Q_VFWD_GAIN being set.
 
-      3) if we are in TRANSITION_TIMER mode then we are transitioning
+      4) if we are in TRANSITION_TIMER mode then we are transitioning
          to forward flight and should put the rotors all the way forward
     */
-    if (plane.control_mode == &plane.mode_qstabilize ||
-        plane.control_mode == &plane.mode_qhover ||
-        plane.control_mode == &plane.mode_qautotune) {
+
+    if (plane.control_mode == &plane.mode_qautotune) {
+        // case 1a
         tiltrotor_slew(0);
         return;
     }
 
+    // if not transitioning to fixed wing and in QACRO, QSTABILIZE or QHOVER mode
+    if (!assisted_flight &&
+        (plane.control_mode == &plane.mode_qacro ||
+         plane.control_mode == &plane.mode_qstabilize ||
+         plane.control_mode == &plane.mode_qhover)) {
```

Comment:
This stuff is already checked in  forward_throttle_pct() I think, so you can probably just call that without checking first. The assisted_flight check is the only one not checked there think

Reply:
Since this returns early, I think that would change behavior.

Reply:
ah, I see what you mean, my mistake